packages/newspringchurchapp/src/content-single/Features/SermonNotes.js

```diff
@@ -0,0 +1,95 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import {
+  H3,
+  PaddedView,
+  ActionCard,
+  Touchable,
+  Icon,
+} from '@apollosproject/ui-kit';
+import share from '../../utils/content/share';
+
+const SermonNotes = ({ contentId, features }) => {
+  const [sharedMsg, changeSharedMsg] = useState('');
+  const [enhancedFeatures, enhanceFeatures] = useState([]);
+  const onNotesChange = (id, text) => {
+    const placeholder = `${id}{{(.*)}}`;
+    const re = new RegExp(placeholder, 'g');
+    changeSharedMsg((msg) => msg.replace(re, `${id}{{${text}}}`));
+  };
+  useEffect(() => {
+    let msg = '';
+    const featuresWithCallbacks = features.map((feature) => {
+      const featureProps = feature.props.children[0].props;
+
+      // assemble starting message without custom notes
+      if (featureProps.sharing) {
+        msg = `${msg + featureProps.sharing.message}\n\n`;
+        return feature;
+      }
+
+      // drop in placeholders for custom notes
+      if (featureProps.id.match(/NoteFeature/g).length > 0)
+        msg = `${msg + featureProps.id}{{}}\n\n`;
+
+      // add callbacks to swap note placeholders with custom text
+      return {
```

Comment:
@redreceipt @cdanwards @vinnyjth @conrad-vanl I've got to admit ... I really have no idea what this section is doing. Help?

Reply:
1. Adds all the non-note features
2. Adds all the placeholders: `NoteFeature123123{{}}`
3. Drops callbacks to update the placeholders inside the note feature components

Reply:
We may have talked about this, but I'm drawing a blank. This seems so overly complex. If someone is sharing, why can't this just go through the features and get the text from each of them. Why is the addition of placeholders and whatever this `return` is doing necessary? Shouldn't this be able to just pull the text from the Note feature and include it as part of the message? What am I missing?